### ISSUE TITLE
FFmpeg Requesting 'iHD' driver for i915(i965?)

### ISSUE THREAD

**Issue by vindicatorr** [intel/libva#436]
I didn't know if this should be submitted to ffmpeg or here.
I am submitting here because my searching led me to https://github.com/Intel-Media-SDK/MediaSDK/issues/1614.

I am using Haswell and going by https://trac.ffmpeg.org/wiki/Hardware/QuickSync (and archlinux wiki), I would be using VA-API with i965.
I have libva-intel-driver 2.4.1 installed and don't think I should be using intel-media-driver 20.2.0 since it's for Broadwell+ (after Haswell). For the sake of testing, I installed intel-media-driver, then removed it when it failed to work.

`$ ffmpeg -loglevel trace -hwaccel qsv -c:v h264_qsv -i <input.mp4> -c:v h264_qsv -an -vf "hwupload" <output.mp4>`
```
[h264_qsv @ 0x608a84b42e40] Error initializing the MFX video decoder: invalid handle (-6)
Error while decoding stream #0:0: Invalid argument                                                                   
cur_dts is invalid st:0 (0) [init:0 i_done:0 finish:0] (this is harmless if it occurs once at the start per stream)
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x608a84b19340] stream 0, sample 583, dts 29127456                                        
[AVHWDeviceContext @ 0x608a84b2a640] Trying to use DRM render node for device 0, with matching kernel driver (i915).
[AVHWDeviceContext @ 0x608a84b2a640] libva: VA-API version 1.8.0                                                     
[AVHWDeviceContext @ 0x608a84b2a640] libva: User requested driver 'iHD'                                              
[AVHWDeviceContext @ 0x608a84b2a640] libva: Trying to open /usr/lib/dri/iHD_drv_video.so                             
[AVHWDeviceContext @ 0x608a84b2a640] libva: va_openDriver() returns -1                                               
[AVHWDeviceContext @ 0x608a84b2a640] Failed to initialise VAAPI connection: -1 (unknown libva error).
Error creating a QSV device                                                                                          
qsv hwaccel requested for input stream #0:0, but cannot be initialized.
```

I also tried forcing the correct driver with `$ export LIBVA_DRIVER_NAME=i965`

**Comment by uartie** [comment]
The ffmpeg `qsv` plugin depends on MediaSDK.  MediaSDK does not support i965 driver.  https://github.com/Intel-Media-SDK/MediaSDK#system-requirements.

You will need to use the ffmpeg `vaapi` plugin, instead.

**Comment by uartie** [comment]
Try:
```
LIBVA_DRIVER_NAME=i965 ffmpeg -hwaccel vaapi \
 -init_hw_device vaapi=hw:/dev/dri/renderD128 \
 -i <input.mp4> -c:v h264_vaapi -an -vf "hwupload" <output.mp4>
```

Also, see https://trac.ffmpeg.org/wiki/Hardware/VAAPI.
If you have further issues, please file with ffmpeg bug tracker.

**Comment by vindicatorr** [comment]
Weird, I could've sworn I had tried that. In fact, I'm sure I did. Maybe it was some combination of options that I was getting wrong.

I knew about that VAAPI link, as that's how I got to the QuickSync page and I was under the impression that it was for Haswell (gen5) as well, given it's listing there.
Maybe, I should post to ffmpeg about that so they can remove Haswell from that list? Or maybe I'm misreading that list into thinking the "-" notation was meant to represent that generation implements all of the support features from the previous generations.
On the flipside, https://trac.ffmpeg.org/ticket/5324 and https://trac.ffmpeg.org/ticket/5732 shows folks trying to use QSV with their Haswell. I just dunno.

Ugh, https://www.reddit.com/r/PleX/comments/bfjlpb/plex_on_linux_how_do_you_tell_whether_youre_using/eleinbz/
So QuickSync is used by either the vaapi or qsv plugin. Methinks the plugin should've just been qsv which handled all implementations. OR instead of naming the one plugin qsv and the other vaapi, qsv plugin should've been named mfx or maybe msdk.

All the same, I thank you for posting it anyway.
